src/twisted/web/test/test_http.py

```diff
@@ -1469,6 +1469,21 @@ def finished(extra):
         self.assertEqual(errors, [])
         self.assertEqual(successes, [True])
 
+    def test_tailerServerTiming(self):
+        """
+        L{_ChunkedTransferDecoder.dataReceived} decodes chunked-encoded data
+        and correctly handle trailer Server-Timing data.
+        """
+        L = []
+        finished = []
+        p = http._ChunkedTransferDecoder(L.append, finished.append)
+        p.dataReceived(b"3\r\nabc\r\n5\r\n12345\r\n")
```

Comment:
To help with test, it might help to also run a separate test in which each byte is received in a separate TCP package.

So, as a testing "strategy"
run a test in which all the data is received in one TCP packet , by using a single `dataReceived` call and another test in which we iterate over the data and call `dataReceive` for each byte.

Here, I can see the data is received in 2 packet.

Reply:
Server-Timing data cannot be split to parts because we don't have any information about the data size. So Server-Timing data should be in a single backet.

Reply:
This isn't how TCP works, though.  There is no guarantee about where packet boundaries will fall.  @adiroiban 's suggestion to write a test that delivers the data one byte at a time represents one extreme end of the *legal* TCP behavior - to split the stream into single byte packets and deliver them separately.

So there must be some framing information *somewhere* to use.  It's just part of the trailer isn't it?  So I would expect that normal trailer framing rules apply.

Reply:
I find out that the topic is a bit wider. It is required to implement "trailing headers" (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer). I thought that trailing header can be only Server-Timing, but it can be almost any type. So I'm going to implement handling of any type of trailing header. And don't know should I give users access to such trailing headers. Or I can just ignore them?

Reply:
Yes... so we have "HTTP chunks" and "TCP packets payload"

The trailing headers can only be delivered via the last HTTP chunk.

But in theory, the single HTTP chunk can be delivered via multiple TCP packets.

----------

In terms of API. `finishCallback` init argument of the `_ChunkedTransferDecoder` is used as a private API.
The value passed by `finishCallback` is handled by `HTTPChannel._finishRequestBody`  which is also a private API.

If I understand correctly, `finishCallback` returns bytes, which might be part of the next response.

Maybe we can update `finishCallback` to call with bytes + trailing headers.

Then `Response.deliverBody` takes an `IProtocol` for which on success the `IProtocol.connectionLost` is called with `t.internet._newclient.ResponseDone`
Maybe we can update `ResponseDone` to have `ResponseDone.trailerHeaders`

So you can read the trailer headers after the response was finalized.

This is just the low-level API... which might be ugly... but it's an idea.

----------

I guess that many implementations don't expect `ResponseDone` to have the trailier headers... so in practice, the headers are ignored.

But someone intersted in ready them, can add special handling.

Note that I'm not an expert in HTTP or HTTP client, so this is just a sugestion that I hope can help to move forward with this PR.

----------

At the same time, to reduce the scope of this PR,
I guess we can also have this PR updates to ignore any trailing headers, and just make sure the response doesn't fails.

We can them consider a separate PR to parse and expose the headers in some way.

Regards